TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/AssetController.php

```diff
@@ -185,29 +194,87 @@ public function relatedNodesAction(AssetInterface $asset)
 
         $usageReferences = $this->assetService->getUsageReferences($asset);
         $relatedNodes = [];
+        $inaccessibleRelations = [];
+
+        $existingSites = $this->siteRepository->findAll();
 
-        /** @var AssetUsageInNodeProperties $usage */
         foreach ($usageReferences as $usage) {
-            $documentNodeIdentifier = $usage->getDocumentNode() instanceof NodeInterface ? $usage->getDocumentNode()->getIdentifier() : null;
-
-            $relatedNodes[$usage->getSite()->getNodeName()]['site'] = $usage->getSite();
-            $relatedNodes[$usage->getSite()->getNodeName()]['documentNodes'][$documentNodeIdentifier]['node'] = $usage->getDocumentNode();
-            $relatedNodes[$usage->getSite()->getNodeName()]['documentNodes'][$documentNodeIdentifier]['nodes'][] = [
-                'node' => $usage->getNode(),
-                'nodeData' => $usage->getNode()->getNodeData(),
-                'contextDocumentNode' => $usage->getDocumentNode(),
-                'accessible' => $usage->isAccessible()
+            $inaccessibleRelation = [
+                'type' => get_class($usage),
+                'usage' => $usage
+            ];
+
+            if (!$usage instanceof AssetUsageInNodeProperties) {
+                $inaccessibleRelations[] = $inaccessibleRelation;
+                continue;
+            }
+
+            try {
+                $nodeType = $this->nodeTypeManager->getNodeType($usage->getNodeTypeName());
+            } catch (NodeTypeNotFoundException $e) {
+                $nodeType = null;
+            }
+            $workspace = $this->workspaceRepository->findByIdentifier($usage->getWorkspaceName());
+            $accessible = $this->domainUserService->currentUserCanReadWorkspace($workspace);
```

Comment:
I see fatal errors here, since `$workspace` is `null` in some cases, still investigating.

Reply:
Haha, a custom implementation overriding `AssetUsageInNodePropertiesStrategy` was the culprit.